src/bin/cql-stress-cassandra-stress/operation/mixed.rs

```diff
@@ -0,0 +1,181 @@
+    add_distribution: Box<dyn Distribution>,
+    operation_ratio: Arc<OperationRatio>,
+    clustering_distribution: Box<dyn Distribution>,
+    current_operation: MixedSubcommand,
+    current_operation_count: usize,
+}
+
+pub struct MixedOperationFactory {
+    session: Arc<Session>,
+    settings: Arc<CassandraStressSettings>,
+    read_statement: PreparedStatement,
+    counter_read_statement: PreparedStatement,
+    write_statement: PreparedStatement,
+    counter_write_statement: PreparedStatement,
+    operation_ratio: Arc<OperationRatio>,
+    workload_factory: RowGeneratorFactory,
+    max_operations: Option<u64>,
+    stats: Arc<ShardedStats>,
+}
+
+impl OperationFactory for MixedOperationFactory {
+    fn create(&self) -> Box<dyn Operation> {
+        let counter_params = self.settings.command_params.counter.as_ref().unwrap();
+        let mixed_params = self.settings.command_params.mixed.as_ref().unwrap();
+
+        Box::new(MixedOperation {
+            session: Arc::clone(&self.session),
+            read_statement: self.read_statement.clone(),
+            counter_read_statement: self.counter_read_statement.clone(),
+            write_statement: self.write_statement.clone(),
+            counter_write_statement: self.counter_write_statement.clone(),
+            workload: self.workload_factory.create(),
+            max_operations: self.max_operations,
+            regular_read_row_validator: EqualRowValidator,
+            counter_read_row_validator: ExistsRowValidator,
+            stats: Arc::clone(&self.stats),
+            non_pk_columns_count: self.settings.column.columns.len(),
+            add_distribution: counter_params.add_distribution.create(),
+            operation_ratio: Arc::clone(&self.operation_ratio),
+            clustering_distribution: mixed_params.clustering.create(),
+            current_operation: MixedSubcommand::Read,
+            current_operation_count: 0,
+        })
+    }
+}
+
+impl MixedOperationFactory {
+    pub async fn new(
+        regular_table_name: &'static str,
+        counter_table_name: &'static str,
+        settings: Arc<CassandraStressSettings>,
+        session: Arc<Session>,
+        workload_factory: RowGeneratorFactory,
+        stats: Arc<ShardedStats>,
+    ) -> Result<Self> {
+        let mixed_params = settings.command_params.mixed.as_ref().unwrap();
+        let read_statement =
+            prepare_read_statement(regular_table_name, &session, &settings).await?;
+        let counter_read_statement =
+            prepare_read_statement(counter_table_name, &session, &settings).await?;
+        let write_statement = WriteOperationFactory::prepare_statement(&session, &settings).await?;
+        let counter_write_statement =
+            CounterWriteOperationFactory::prepare_statement(&session, &settings).await?;
+        let max_operations = settings.command_params.common.operation_count;
+        let operation_ratio = Arc::new(mixed_params.operation_ratio.clone());
```

Comment:
I'm not a fan of the approach... You are exposing some details of the existing operations and use them to implement `MixedOperation`.

I think that the operations should be self-contained. `MixedOperation` should just encapsulate them and not care about their internals apart from the fact that they implement some trait. Would it be possible to implement it like that?

Reply:
I thought about introducing some trait, but don't we end up with the same performance issue as in https://github.com/scylladb/cql-stress/pull/34 (`async-trait` boxed futures)?

I haven't compared both of the approaches when it comes to performance, but to me, it looks exactly the same - we end up with per-operation allocation from boxing the futures. I can still do the comparison, if necessary.

Reply:
I see. AFAIK `async` in trait methods is going to be stabilized in Rust 1.75, so maybe we can use `async_trait` until that happens.

Reply:
> I see. AFAIK `async` in trait methods is going to be stabilized in Rust 1.75, so maybe we can use `async_trait` until that happens.

They indeed got stabilized, but unfortunately such traits are not object-safe, and so cannot be used in trait objects. I think the issue still holds. Do you still think we should use `async_trait`?